pilot/pkg/xds/discovery_test.go

```diff
@@ -38,9 +38,11 @@ func createProxies(n int) []*Connection {
 	proxies := make([]*Connection, 0, n)
 	for p := 0; p < n; p++ {
 		proxies = append(proxies, &Connection{
-			conID:       fmt.Sprintf("proxy-%v", p),
-			pushChannel: make(chan *Event),
-			stream:      &fakeStream{},
+			BaseConnection: BaseConnection{
+				conID:       fmt.Sprintf("proxy-%v", p),
+				pushChannel: make(chan any),
```

Comment:
we could consider making BaseConnection generic.. not sure how messy that ends up being

Reply:
Not worth having generics with two type instances. Composition is simpler.

Reply:
I meant on BaseConnection, and still embeddd that. So pushChannel can be `chan T` instead of `chan any`

Reply:
Yes, I see. That would make `ConnectionContext` parametrized too, and every method that uses it. It's possible, but would be quite verbose since most of these public functions would be type-parametric.